doc/source/pipelines.rst

```diff
@@ -0,0 +1,115 @@
+Pipelines
+=========
+
+.. ipython:: python
+   :suppress:
+
+   from pims.tests.test_common import save_dummy_png, clean_dummy_png
+   filenames = ['img-{0}.png'.format(i) for i in range(9)]
+   save_dummy_png('.', filenames, (256, 256, 3))
+   from pims import ImageSequence
+   video = ImageSequence('img-*.png')
+
+   def _as_grey(frame):
+       red = frame[:, :, 0]
+       green = frame[:, :, 1]
+       blue = frame[:, :, 2]
+       return 0.2125 * red + 0.7154 * green + 0.0721 * blue
+
+   from slicerator import pipeline
+   as_grey = pipeline(_as_grey)
+
+Videos loaded by pims are (``FramesSequence`` objects) are like lists of numpy
+arrays. Unlike Python lists of arrays they are "lazy", they only load the data
+from the harddrive when it is necessary.
+
+In order to modify a ``FramesSequence``, for instance to convert RGB color
+videos to grayscale, one could load all the video frames in memory and do the
+conversion. THis however costs a lot of time and memory, and for very large
+videos this is just not feasible. To solve this problem, PIMS uses
+so-called pipeline decorators from a sister project called ``slicerator``.
+A pipeline-decorated function is only evaluated when needed, so that the
+underlying video data is only accessed one element at a time.
+
+Conversion to greyscale
+-----------------------
+
+Say we want to convert an RGB video to greyscale. We define a function as
+follows and decorate it with ``@pipeline`` to turn it into a pipeline:
+
+.. code-block:: python
+
+   from slicerator import pipeline  # or: from pims import pipeline
+
+   @pipeline
+   def as_grey(frame):
+       red = frame[:, :, 0]
+       green = frame[:, :, 1]
+       blue = frame[:, :, 2]
+       return 0.2125 * red + 0.7154 * green + 0.0721 * blue
+
+
+The behavior of ``as_grey`` is unchanged if it is used on a single frame:
+
+.. ipython:: python
+
+   frame = video[0]
+   print(frame.shape)   # the shape of the example video is RGB
+   processed_frame = as_grey(video[0])
+   print(processed_frame.shape)  # the converted frame is indeed greyscale
+
+
+However, the ``@pipeline`` decorator enables lazy evaluation of full videos:
```

Comment:
I think you want to make it explicit that the modified video can be used exactly as you would use the original pims object. In most cases, it should now look as though you are accessing a grayscale video file, even though the file on disk is still in color.